searchcore/src/vespa/searchcore/proton/server/searchview.cpp

```diff
@@ -161,8 +161,9 @@ SearchView::getDocsumsInternal(const DocsumRequest & req)
 }
 
 std::unique_ptr<SearchReply>
-SearchView::match(const ISearchHandler::SP &self, const SearchRequest &req, ThreadBundle &threadBundle) const {
-    return _matchView->match(self, req, threadBundle);
+SearchView::match(const SearchRequest &req, ThreadBundle &threadBundle) const {
+    ISearchHandler::SP self = const_cast<SearchView *>(this)->shared_from_this();
```

Comment:
I do not like this const-cast. Maybe this function should not be const (since it is allowed to change the object state).

Reply:
Const cast was not necessary. Just passing a const ISearchHandler was sufficient.